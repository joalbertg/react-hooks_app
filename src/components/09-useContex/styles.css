body {
  background-color: black;
  color: white;
}

hr {
  border-top: 1px solid rgba(255, 255, 255, .5);
}

._404 {
  display: flex;
  height: calc(100vh - 20rem);
  justify-content: center;
  align-items: center;
  font-size: 4rem;
}

pre {
  color: white;
}

