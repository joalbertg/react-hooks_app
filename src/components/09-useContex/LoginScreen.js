import React, { useContext } from 'react';

import { UserContext } from './UserContext';

import './styles.css';

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  const handleUser = () => {
    setUser({
      id: new Date().getTime(),
      name: 'joalbert',
      email: 'joalbertgonzalez@gmail.com'
    });
  }

  return(
    <>
      <h1>LoginScreen</h1>
      <hr />

      <button
        className="btn btn-info"
        onClick={handleUser}
      >
        Login
      </button>
    </>
  );
}
