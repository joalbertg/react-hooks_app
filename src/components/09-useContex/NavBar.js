import React from 'react';
import { Link, NavLink } from 'react-router-dom';

import './styles.css';

export const NavBar = () => {
  return(
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <Link
        className="navbar-brand"
        to="/"
      >
          useContex
      </Link>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarSupportedContent"
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation">
        <span className="navbar-toggler-icon"></span>
      </button>

      <div className="collapse navbar-collapse" id="navbarSupportedContent">
        <ul className="navbar-nav mr-auto">
          {/*
          <li className="nav-item">
            <Link className="nav-link" to="/">Home</Link>
          </li>
          <li className="nav-item">
            <Link className="nav-link" to="/about">About</Link>
          </li>
          <li className="nav-item">
            <Link className="nav-link" to="/login">Login</Link>
          </li>
          */}

          { /* para poder usar la clase active,
               el Link y el NavLink son exactamente iguales,
               solo que con el NavLink se puede establecer una 
               clase cuando el link corresponde
          */ }
          <li className="nav-item">
            <NavLink exact activeClassName="active" className="nav-link" to="/">Home</NavLink>
          </li>
          <li className="nav-item">
            <NavLink exact activeClassName="active" className="nav-link" to="/about">About</NavLink>
          </li>
          <li className="nav-item">
            <NavLink exact activeClassName="active" className="nav-link" to="/login">Login</NavLink>
          </li>

        </ul>
      </div>
    </nav>
  );
}

