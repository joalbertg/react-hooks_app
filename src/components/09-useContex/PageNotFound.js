import React from 'react';

import './styles.css';

export const PageNotFound = () => {
  console.log('++++')
  return(
    <div className="_404">
      Error: 404 Page Not Found!!!
    </div>
  );
}

