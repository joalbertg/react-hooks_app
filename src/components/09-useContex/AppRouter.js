import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from "react-router-dom";

import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { PageNotFound } from './PageNotFound';
import { NavBar } from './NavBar';

export const AppRouter = () => {
  return(
    <Router>
      <>
        <NavBar />
        <div className="container">
          <Switch>
            <Route exact path="/" component={HomeScreen} />
            <Route exact path="/about" component={AboutScreen} />
            <Route exact path="/login" component={LoginScreen} />
            <Route exact path="/404" component={PageNotFound} />
            <Redirect to="/404" />
          </Switch>
        </div>
      </>
    </Router>
  );
}

