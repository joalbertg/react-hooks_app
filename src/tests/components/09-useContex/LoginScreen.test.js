import React from 'react';
import { mount } from 'enzyme';

import { LoginScreen } from '../../../components/09-useContex/LoginScreen';
import { UserContext } from '../../../components/09-useContex/UserContext';

describe('Tests LoginScreen component', () => {
  const setUser = jest.fn();
  const user = {
    name: 'joalbert',
    email: 'joalbertgonzalez@gmail.com'
  };

  const wrapper = mount(
    <UserContext.Provider value={ {setUser} }>
      <LoginScreen />
    </UserContext.Provider>
  );

  test('Should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should call setUser function', () => {
    wrapper.find('button').prop('onClick')();

    expect(setUser).toHaveBeenCalled();
    expect(setUser).toHaveBeenCalledWith({
      id: expect.any(Number),
      ...user
    });
  });
});

