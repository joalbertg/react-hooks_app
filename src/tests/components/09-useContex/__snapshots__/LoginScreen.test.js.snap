// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests LoginScreen component Should display correctly 1`] = `
Array [
  <h1>
    LoginScreen
  </h1>,
  <hr />,
  <button
    className="btn btn-info"
    onClick={[Function]}
  >
    Login
  </button>,
]
`;
